### DISCUSSION TITLE
How to use with latest Tonic?

### DISCUSSION THREAD

**Discussion by timvw01** [devashishdxt/tonic-web-wasm-client#21] [Q&A] [answered]
Hi

I am trying to make this work with the Tonic json-codec example.

Tonic generates a Client struct that is initialized with Client::connect(url), resulting in a Client<Channel> type.

How should this be used with the tonic_web_wasm_client::client?

Is it possible to post a working example that people can follow?

Thanks.

**Comment by devashishdxt** [comment, marked as answer]
Here's the example in `test-suite` folder: https://github.com/devashishdxt/tonic-web-wasm-client/tree/main/test-suite

**Comment by timvw01** [comment]
So is it correct to assume that this package only works with a proto file? I.e tonic generated files build with tonic_build::manual are not supported?

**Reply by devashishdxt** [reply]
I'm not sure what you're trying to do with `tonic_build::manual`. Can you share your code?

**Reply by timvw01** [reply]
Thanks for your replies. In the meantime i got the basic setup to work. Right now trying to work out how to use TLS..

Background just in case: Our project uses flexbuffers instead of protobufs. To use tonic with flexbuffers, you can build the tonic grpc client and server with tonic_build::manual. 

A similiar thing is done in the examples dir of Tonic, but then with Json instead of Flex. The example is found here:

https://github.com/hyperium/tonic/tree/master/examples/src/json-codec

The build command that is used is:

fn build_json_codec_service() {
    let greeter_service = tonic_build::manual::Service::builder()
        .name("Greeter")
        .package("json.helloworld")
        .method(
            tonic_build::manual::Method::builder()
                .name("say_hello")
                .route_name("SayHello")
                .input_type("crate::common::HelloRequest")
                .output_type("crate::common::HelloResponse")
                .codec_path("crate::common::JsonCodec")
                .build(),
        )
        .build();
    tonic_build::manual::Builder::new().compile(&[greeter_service]);
}

The resulting client is used as follows:

let mut client = GreeterClient::connect("http://[::1]:50051").await?;

I got confused at the ::connect part. It uses Tonic transport, which is not compatible with wasm32. 

tonic-web-wasm-client is an excellent package, seems to me that this functionality should be part of the official Tonic.  If there is a TLS example available somewhere that would be great.

Thanks.

**Reply by devashishdxt** [reply]
There's already a PR with working TLS example: https://github.com/devashishdxt/tonic-web-wasm-client/pull/14/files
This PR cannot be merged because on CI, we cannot configure browser to accept self-signed certificates.
